src/Adapter/Preferences/PreferencesConfiguration.php

```diff
@@ -125,22 +118,28 @@ protected function validateSameSiteConfiguration(array $configuration): bool
     }
 
     /**
-     * {@inheritdoc}
+     * @return OptionsResolver
      */
-    public function validateConfiguration(array $configuration)
+    protected function buildResolver(): OptionsResolver
     {
-        return isset(
-            $configuration['enable_ssl'],
-            $configuration['enable_ssl_everywhere'],
-            $configuration['enable_token'],
-            $configuration['allow_html_iframes'],
-            $configuration['use_htmlpurifier'],
-            $configuration['price_round_mode'],
-            $configuration['price_round_type'],
-            $configuration['display_suppliers'],
-            $configuration['display_manufacturers'],
-            $configuration['display_best_sellers'],
-            $configuration['multishop_feature_active']
-        );
+        $resolver = (new OptionsResolver())
+            ->setDefined(
```

Comment:
Maybe you should add the allowed types for each field?

Reply:
done 👍